src/lib/io/sol-iio.c

```diff
@@ -43,6 +43,7 @@
 #endif
 
 struct sol_iio_device {
+    double *mount_matrix;
```

Comment:
Please check on the line 787, if there is no such mount_matrix(get_mount_marix return false), the pointer will be free